src/EthersSafe.ts

```diff
@@ -112,97 +159,138 @@ class EthersSafe implements Safe {
   }
 
   /**
-   * Signs data using the current owner account.
+   * Signs a hash using the current signer account.
    *
-   * @param hash - The data to sign
+   * @param hash - The hash to sign
    * @returns The Safe signature
    */
   async signTransactionHash(hash: string): Promise<SafeSignature> {
-    const address = await this.#signer.address
+    if (!this.#signer) {
+      throw new Error('No signer provided')
+    }
+    const owners = await this.getOwners()
+    if (
+      !owners.find(
+        (owner: string) => this.#signer && areAddressesEqual(owner, this.#signer.address)
+      )
+    ) {
+      throw new Error('Transactions can only be signed by Safe owners')
+    }
     const messageArray = this.#ethers.utils.arrayify(hash)
     const signature = await this.#signer.signMessage(messageArray)
-    return new EthSignSignature(address, signature)
+    return new EthSignSignature(this.#signer.address, signature)
   }
 
   /**
-   * Adds the signature of the current owner to the Safe transaction object.
+   * Adds the signature of the current signer to the Safe transaction object.
    *
    * @param safeTransaction - The Safe transaction to be signed
    */
-  async confirmTransaction(safeTransaction: SafeTransaction): Promise<void> {
-    const owners = await this.getOwners()
-    if (!owners.find((owner: string) => areAddressesEqual(owner, this.#signer.address))) {
-      throw new Error('Transactions can only be confirmed by Safe owners')
-    }
+  async signTransaction(safeTransaction: SafeTransaction): Promise<void> {
     const txHash = await this.getTransactionHash(safeTransaction)
     const signature = await this.signTransactionHash(txHash)
     safeTransaction.signatures.set(signature.signer, signature)
   }
 
   /**
-   * Returns the encoding of a Safe transaction.
+   * Approves on-chain a hash using the current signer account.
    *
-   * @param transaction - The Safe transaction
-   * @returns The encoding of the Safe transaction
+   * @param hash - The hash to approve
+   * @param skipOnChainApproval - TRUE to avoid the Safe transaction to be approved on-chain
+   * @returns The pre-validated signature
    */
-  async encodeTransaction(transaction: SafeTransaction): Promise<string> {
-    const encodedTx = await this.#contract.interface.encodeFunctionData('execTransaction', [
-      transaction.data.to,
-      transaction.data.value,
-      transaction.data.data,
-      transaction.data.operation,
-      transaction.data.safeTxGas,
-      transaction.data.baseGas,
-      transaction.data.gasPrice,
-      transaction.data.gasToken,
-      transaction.data.refundReceiver,
-      transaction.encodedSignatures()
-    ])
-    return encodedTx
+  async approveTransactionHash(
+    hash: string,
+    skipOnChainApproval?: boolean
+  ): Promise<SafeSignature> {
+    if (!this.#signer) {
+      throw new Error('No signer provided')
+    }
+    const owners = await this.getOwners()
```

Comment:
Should we do this check also for the `signTransactionHash` method?

Reply:
It's already there https://github.com/gnosis/safe-core-sdk/pull/16/files#diff-52e06b30c111e17c1b5f4d35c386e257bf013445d5354b6acaccacfca904eb6bR168